### PR TITLE
[10.0] migrate delivery_carrier_label postlogistics

### PR DESCRIPTION
**Pull request by yvaucher**
cherry pick of commits from 9.0

Plus

* Simplify signature of soap connector by removing company
* As carrier inherits from product, change postlogistics partner as
  a service product

Specific changes to v 10.0 are in https://github.com/yvaucher/carrier-delivery/commit/abfa3395b1666f41b7673f4f24d8878a6e5f2dd3

### CODE REVIEW

File: delivery_carrier_label_postlogistics/models/delivery.py [hunk 1]
```diff
@@ -105,29 +105,25 @@ class DeliveryCarrierOption(models.Model):
...
+        if default_carrier_id:
+            carrier = self.env['delivery.carrier'].browse(default_carrier_id)
+            if carrier.is_postlogistics():
                 arch = result['arch']
                 doc = etree.fromstring(arch)
                 for node in doc.xpath("//field[@name='tmpl_option_id']"):
                     node.set(
                         'domain',
-                        "[('partner_id', '=', %s), "
+                        "[('product_id', '=', %s), "
```
Comment (grindtildeath) [inline delivery_carrier_label_postlogistics/models/delivery.py]: There is no `product_id` on `delivery.carrier.template.option` model, what brings :
```
ValueError: Invalid field u'product_id' in leaf "<osv.ExtendedLeaf: (u'product_id', u'=', 2) on delivery_carrier_template_option (ctx: )>"
```

**Review by gurneyalex — APPROVED** [review]

File: delivery_carrier_label_postlogistics/__manifest__.py [hunk 2]
```diff
@@ -2,23 +2,26 @@
...
  'author': "Camptocamp,Odoo Community Association (OCA)",
  'maintainer': 'Camptocamp',
  'license': 'AGPL-3',
  'category': 'Delivery',
  'complexity': 'normal',
- 'depends': ['base_delivery_carrier_label'],
+ 'depends': ['base_delivery_carrier_label',
+             'configuration_helper',
+             'web_domain_field'],
  'website': 'http://www.camptocamp.com/',
```
Comment (simahawk) [inline delivery_carrier_label_postlogistics/__manifest__.py]: ttps://github.com/OCA/delivery-carrier

File: delivery_carrier_label_postlogistics/data/partner.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<odoo>
+  <record id="partner_postlogistics" model="res.partner">
```
Comment (simahawk) [inline delivery_carrier_label_postlogistics/data/partner.xml]: noupdate?

File: delivery_carrier_label_postlogistics/models/company.py [hunk 4]
```diff
@@ -51,11 +53,28 @@ class ResCompany(models.Model):
...
+         ('picking_num', 'Use picking number with pack counter')],
+        string="Tracking number format",
+        default='postlogistics',
+    )
+    postlogistics_proclima_logo = fields.Boolean('Print ProClima logo')
 
-    def _get_wsdl_url(self):
+    @api.depends('postlogistics_test_mode')
+    def _compute_wsdl_url(self):
+        path = 'delivery_carrier_label_postlogistics/data/'
```
Comment (simahawk) [inline delivery_carrier_label_postlogistics/models/company.py]: would be better to use `from odoo.modules.module import get_resource_path`

File: delivery_carrier_label_postlogistics/README.rst [hunk 5]
```diff
@@ -23,6 +23,15 @@ As a requirement you need to install `suds-jurko` library. It will fail with the
 latest and outdated version of `suds`.
 https://pypi.python.org/pypi/suds-jurko/0.6
```
Comment (simahawk) [inline delivery_carrier_label_postlogistics/README.rst]: can you update whole readme based on new layout? https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

**Review by simahawk — CHANGES_REQUESTED** [review]
small things

**Comment by grindtildeath** [comment]
@simahawk Fixed, and squashed my last commits. I left @yvaucher 's as they are.

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
@grindtildeath once you fix this warning on runbot we should be able to merge /cc @gurneyalex @yvaucher 
```
2018-05-29 11:16:29,382 182 WARNING openerp_test odoo.addons.base.module.module: module delivery_carrier_label_postlogistics: description is empty !
```

**Comment by grindtildeath** [comment]
@simahawk Shouldn't we backport this commit ? https://github.com/OCA/delivery-carrier/pull/155/commits/64e34224d6754e5cb65e3ad8eb73b02526b76545

**Comment by simahawk** [comment]
@grindtildeath I'd say "yes". See my comments on it ;)

**Pull request merged**